plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1290,6 +1257,243 @@ public void testJoinPushdown()
+                .collect(toImmutableList());
+
+        // Disable DF here for the sake of negative test cases' expected plan. With DF enabled, some operators return in DF's FilterNode and some do not.
+        Session withoutDynamicFiltering = Session.builder(session)
+                .setSystemProperty("enable_dynamic_filtering", "false")
+                .build();
+
+        // inequality
+        for (String operator : nonEqualities) {
+            // bigint inequality predicate
+            assertThat(query(withoutDynamicFiltering, format("SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey %s r.regionkey", operator)))
+                    // Currently no pushdown as inequality predicate is removed from Join to maintain Cross Join and Filter as separate nodes
+                    .isNotFullyPushedDown(broadcastJoinOverTableScans);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n LEFT JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator),
+                    broadcastJoinOverTableScansRepartition);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n RIGHT JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator),
+                    broadcastJoinOverTableScansRepartition);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n FULL JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator) && hasBehavior(SUPPORTS_JOIN_PUSHDOWN_WITH_FULL_JOIN),
+                    broadcastJoinOverTableScansRepartition);
+        }
+
+        // IS DISTINCT FROM
+        String distinctOperator = "IS DISTINCT FROM";
+        assertConditionallyPushedDown(
+                withoutDynamicFiltering,
+                format("SELECT r.name, n.name FROM nation n LEFT JOIN region r ON n.regionkey %s r.regionkey", distinctOperator),
+                expectJoinPushdown(distinctOperator),
+                broadcastJoinOverTableScans);
+
+        assertConditionallyPushedDown(
+                withoutDynamicFiltering,
+                format("SELECT r.name, n.name FROM nation n RIGHT JOIN region r ON n.regionkey %s r.regionkey", distinctOperator),
+                expectJoinPushdown(distinctOperator),
+                broadcastJoinOverTableScansRepartition);
+
+        assertConditionallyPushedDown(
+                withoutDynamicFiltering,
+                format("SELECT r.name, n.name FROM nation n FULL JOIN region r ON n.regionkey %s r.regionkey", distinctOperator),
+                expectJoinPushdown(distinctOperator) && hasBehavior(SUPPORTS_JOIN_PUSHDOWN_WITH_FULL_JOIN),
```

Comment:
considering comments above expectJoinPushdown could have `joinOperator` parameter

Reply:
and use `assertConditionallyPushedDown` with multiple test parameters, prepared by crossjoinin data providers, I know @kokosing you don't like this approach too much, but there is a lot of data tested here in very similar way, and seems it would not be hard to track data providers generation.

Reply:
simplified